server/pulp/plugins/conduits/mixins.py

```diff
@@ -305,40 +305,33 @@ def __init__(self, repo_id, distributor_id):
 
     def get_scratchpad(self):
         """
-        Returns the value set in the scratchpad for this repository. If no
-        value has been set, None is returned.
-
-        @return: value saved for the repository and this distributor
-        @rtype:  <serializable>
+        Returns the scratchpad for this Distributor or None if it has not been set.
 
-        @raises DistributorConduitException: wraps any exception that may occur
-                in the Pulp server
+        :return: value saved for the repository and this distributor
+        :rtype:  dict or None
+        :raises DistributorConduitException: wraps any exception that may occur in the Pulp server
         """
         try:
-            distributor_manager = manager_factory.repo_distributor_manager()
-            value = distributor_manager.get_distributor_scratchpad(self.repo_id,
-                                                                   self.distributor_id)
-            return value
+            qs = model.Distributor.objects.get_or_404
+            distributor = qs(repo_id=self.repo_id, distributor_id=self.distributor_id)
+            return distributor.scratchpad or None
```

Comment:
Gotcha. The new behavior is slightly different than the old. In the previous case, if the scratchpad was an empty dict, ``get_scratchpad()`` would return an empty dict. In your case, it would return None instead.

The difference is that calling ``.get('scratchpad', None)`` only returns the default None if the key ``'scratchpad'`` doesn't exist on the document. In your code, the value for ``.scratchpad`` is evaluated as a ``bool`` and replaced with ``None`` if that evaluates to ``False``.

Does that make sense?